src/file_name_plugin/prompt.js

```diff
@@ -95,10 +95,12 @@ export default class FileNamePatternPrompt extends PatternPrompt {
     if (regex) {
       this._searchSources.forEach(({ testPaths, config }) => {
         tests = tests.concat(
-          testPaths.filter(testPath => testPath.match(pattern)).map(path => ({
-            path,
-            context: { config },
-          })),
+          testPaths
+            .filter(testPath => (regex ? testPath.match(regex) : false))
```

Comment:
can use `.test` as you only care about true/false, not the match

Reply:
also, `regex` will always be truthy as this is inside `if (regex)`

Reply:
Yeah, but Flow doesn't know that https://flow.org/en/docs/lang/refinements/#toc-refinement-invalidations

Although I think I can rewrite it a bit to avoid this.